### PR TITLE
Redant directory structure enhanced

### PR DESCRIPTION
**Pull request by nishith-vihar**
directory structure has been enhanced and dependencies
resolved

Fixes: #116
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
The directory structure is as follows in case of support,

```
support
│   └── redant_libs
│       ├── gluster_libs
│       │   ├── gluster_ops.py
│       │   ├── __init__.py
│       │   ├── peer_ops.py
│       │   └── volume_ops.py
│       ├── redant_mixin.md
│       ├── redant_mixin.py
│       └── support_libs
│           ├── __init__.py
│           ├── io_ops.py
│           ├── logging.md
│           ├── relog.py
│           └── rexe.py
```

Why do we have redant_libs in between ? Can't we directly have everything under support ?

File: tests/example/test_sample.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 hold the functions to be run in the test case.
 """
 
-#disruptive;distributed,replicated,ariter,dispersed
+# disruptive;distributed,replicated,ariter,dispersed
```
Comment (srijan-sivakumar) [inline tests/example/test_sample.py:7]: Remove space. This is not a regular comment. A single space added here will effect the test_list_builder to parse for the space everytime for every TCs.

**Comment by nishith-vihar** [comment]
> Why do we have redant_libs in between ? Can't we directly have everything under support ?
I thought that support dir could contain some other support files. So i thought i will keep all the libs under redant_libs as it was before.

File: tests/example/test_sample.py (hunk 1 above)
Comment (nishith-vihar) [reply tests/example/test_sample.py:7]: I will make that change

**Comment by BarakSason** [comment]
> The directory structure is as follows in case of support,
> 
> ```
> support
> │   └── redant_libs
> │       ├── gluster_libs
> │       │   ├── gluster_ops.py
> │       │   ├── __init__.py
> │       │   ├── peer_ops.py
> │       │   └── volume_ops.py
> │       ├── redant_mixin.md
> │       ├── redant_mixin.py
> │       └── support_libs
> │           ├── __init__.py
> │           ├── io_ops.py
> │           ├── logging.md
> │           ├── relog.py
> │           └── rexe.py
> ```
> 
> Why do we have redant_libs in between ? Can't we directly have everything under support ?

I don't completely agree with this structure (have support dir and "support libs" inside...).
The following seems more natural to me (all the following is inside "support" dir):
ops (dir) - all ops files inside
logging (dir) - All log related stuff inside
mixin (dir) - all mixing stuff inside
distributed executioner - all rexe stuff inside

What do you think?

**Comment by srijan-sivakumar** [comment]
> > The directory structure is as follows in case of support,
> > ```
> > support
> > │   └── redant_libs
> > │       ├── gluster_libs
> > │       │   ├── gluster_ops.py
> > │       │   ├── __init__.py
> > │       │   ├── peer_ops.py
> > │       │   └── volume_ops.py
> > │       ├── redant_mixin.md
> > │       ├── redant_mixin.py
> > │       └── support_libs
> > │           ├── __init__.py
> > │           ├── io_ops.py
> > │           ├── logging.md
> > │           ├── relog.py
> > │           └── rexe.py
> > ```
> > 
> > 
> > Why do we have redant_libs in between ? Can't we directly have everything under support ?
> 
> I don't completely agree with this structure (have support dir and "support libs" inside...).
> The following seems more natural to me (all the following is inside "support" dir):
> ops (dir) - all ops files inside
> logging (dir) - All log related stuff inside
> mixin (dir) - all mixing stuff inside
> distributed executioner - all rexe stuff inside
> 
> Edit - I also don't think the "redant" prefix is needed at all regarding file names.
> 
> What do you think?

Ack. I think the classification acc to
1. ops
2. rexe
3. logging
4. mixin
is good. All these coming under support and all ops coming under ops. It'd be easier to add new modules tomorrow in this manner. 

One things which can be a little different is mixin instead of being put inside a directory can be directly put under support. Logically and even representationally it'll be then inheriting from all the support modules.

And secondly, yeah. It is a little annoying while testing out the changes as these scripts have the same redant_ prefix.

**Comment by BarakSason** [comment]
> > > The directory structure is as follows in case of support,
> > > ```
> > > support
> > > │   └── redant_libs
> > > │       ├── gluster_libs
> > > │       │   ├── gluster_ops.py
> > > │       │   ├── __init__.py
> > > │       │   ├── peer_ops.py
> > > │       │   └── volume_ops.py
> > > │       ├── redant_mixin.md
> > > │       ├── redant_mixin.py
> > > │       └── support_libs
> > > │           ├── __init__.py
> > > │           ├── io_ops.py
> > > │           ├── logging.md
> > > │           ├── relog.py
> > > │           └── rexe.py
> > > ```
> > > 
> > > 
> > > Why do we have redant_libs in between ? Can't we directly have everything under support ?
> > 
> > 
> > I don't completely agree with this structure (have support dir and "support libs" inside...).
> > The following seems more natural to me (all the following is inside "support" dir):
> > ops (dir) - all ops files inside
> > logging (dir) - All log related stuff inside
> > mixin (dir) - all mixing stuff inside
> > distributed executioner - all rexe stuff inside
> > Edit - I also don't think the "redant" prefix is needed at all regarding file names.
> > What do you think?
> 
> Ack. I think the classification acc to
> 
>     1. ops
> 
>     2. rexe
> 
>     3. logging
> 
>     4. mixin
>        is good. All these coming under support and all ops coming under ops. It'd be easier to add new modules tomorrow in this manner.
> 
> 
> One things which can be a little different is mixin instead of being put inside a directory can be directly put under support. Logically and even representationally it'll be then inheriting from all the support modules.
> 
> And secondly, yeah. It is a little annoying while testing out the changes as these scripts have the same redant_ prefix.

OK with me.

**Comment by srijan-sivakumar** [comment]
Did you run the tests after making these changes ?

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**